hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -100,6 +141,338 @@ func (r *NodePoolReconciler) reconcileInPlaceUpgrade(ctx context.Context, hc *hy
+	log := ctrl.LoggerFrom(ctx)
+
+	desiredVerb := strings.Split(desiredState, "-")[0]
+	if desiredVerb == DrainerStateUncordon {
+		// TODO delete the pod after we uncordon
+	}
+
+	// TODO actually implement the node draining. For now, just set the singal and pretend we drained.
+	annos := map[string]string{
+		LastAppliedDrainerAnnotationKey: desiredState,
+	}
+	if result, err := controllerutil.CreateOrPatch(ctx, hostedClusterClient, node, func() error {
+		return r.reconcileNodeAnnotations(
+			ctx,
+			node,
+			annos,
+		)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile node drain annotations: %w", err)
+	} else {
+		log.Info("Reconciled Node drain annotations", "result", result)
+	}
+	return nil
+}
+
+func (r *NodePoolReconciler) reconcileNodeAnnotations(ctx context.Context, node *corev1.Node, annos map[string]string) error {
+	for k, v := range annos {
+		node.Annotations[k] = v
+	}
+	return nil
+}
+
+func (r *NodePoolReconciler) createInPlaceUpdateManifests(ctx context.Context, hostedClusterClient client.Client, targetConfigVersionHash, ignEndpoint string, caCertBytes, tokenBytes []byte, nodePool *hyperv1.NodePool) error {
+	// TODO properly create the objects
+	// TODO use updated CreateOrUpdate func
+	// TODO un-hardcode objects
+	log := ctrl.LoggerFrom(ctx)
+
+	namespace := inPlaceUpgradeNamespace(nodePool)
+
+	if _, err := controllerutil.CreateOrPatch(ctx, hostedClusterClient, namespace, nil); err != nil {
+		return fmt.Errorf("failed to reconcile update namespace: %v", err)
+	}
+
+	configmap := inPlaceUpgradeConfigMap(nodePool, namespace.Name)
+
+	if result, err := controllerutil.CreateOrPatch(ctx, hostedClusterClient, configmap, func() error {
+		return r.reconcileUpgradeConfigmap(
+			ctx,
+			configmap,
+			targetConfigVersionHash, ignEndpoint,
+			caCertBytes, tokenBytes,
+		)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile upgrade ConfigMap for hash %s: %w", targetConfigVersionHash, err)
+	} else {
+		log.Info("Reconciled ConfigMap", "result", result)
+	}
+	return nil
+}
+
+func (r *NodePoolReconciler) reconcileUpgradeConfigmap(ctx context.Context,
+	configmap *corev1.ConfigMap,
+	targetConfigVersionHash, ignEndpoint string,
+	caCertBytes, tokenBytes []byte) error {
+
+	log := ctrl.LoggerFrom(ctx)
+	// fetch desired config off our ign endpoint and then stuff into configmap
+	// TODO reconsider this workflow
+	ignURL := fmt.Sprintf("https://%s/ignition", ignEndpoint)
+	req, err := http.NewRequest("GET", ignURL, nil)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to construct request")
+	}
+	req.Header.Add("Accept", "application/vnd.coreos.ignition+json;version=3.2.0, */*;q=0.1")
+	encodedToken := base64.StdEncoding.EncodeToString(tokenBytes)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", encodedToken))
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCertBytes)
+	client := &http.Client{
+		Transport: &http.Transport{
```

Comment:
I'd split this in two funcs: fetchTokenConfig or so, the reconcileUpgradeConfigmap just receive the content in the signature

Reply:
Reworded TODO above